autotest/pyscripts/test_ogr_layer_algebra.py

```diff
@@ -0,0 +1,489 @@
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import pytest
+import test_py_scripts
+
+from osgeo import ogr
+
+###############################################################################
+
+# Test Intersection
+
+
+def test_ogr_layer_algebra_1():
+
+    script_path = test_py_scripts.get_py_script("ogr_layer_algebra")
+    if script_path is None:
+        pytest.skip()
+
+    # Create input,method,output paths for intersection.
+    input_path = "tmp/input_layer.shp"
+    method_path = "tmp/method_layer.shp"
+    output_path = "tmp/output_layer.shp"
+
+    # definition of input,method layers
+    input_layer = ogr.GetDriverByName("ESRI Shapefile").CreateDataSource(input_path)
+    method_layer = ogr.GetDriverByName("ESRI Shapefile").CreateDataSource(method_path)
+
+    A = input_layer.CreateLayer("poly")
+    B = method_layer.CreateLayer("poly")
+
+    a1 = "POLYGON((1 2, 1 3, 3 3, 3 2, 1 2))"
+    a2 = "POLYGON((5 2, 5 3, 7 3, 7 2, 5 2))"
+    b1 = "POLYGON((2 1, 2 4, 6 4, 6 1, 2 1))"
+
+    feat = ogr.Feature(A.GetLayerDefn())
+    feat.SetGeometryDirectly(ogr.Geometry(wkt=a1))
+    A.CreateFeature(feat)
+
+    feat = ogr.Feature(A.GetLayerDefn())
+    feat.SetGeometryDirectly(ogr.Geometry(wkt=a2))
+    A.CreateFeature(feat)
+
+    input_layer = None
+
+    feat = ogr.Feature(B.GetLayerDefn())
+    feat.SetGeometryDirectly(ogr.Geometry(wkt=b1))
+    B.CreateFeature(feat)
+
+    method_layer = None
+
+    # executing script
+    test_py_scripts.run_py_script(
+        script_path,
+        "ogr_layer_algebra",
+        f"Intersection -input_ds {input_path} -output_ds {output_path}  -method_ds {method_path}",
+    )
+
+    driver = ogr.GetDriverByName("ESRI Shapefile")
+    dataSource = driver.Open(output_path, 0)
+    layer = dataSource.GetLayer()
+    featureCount = layer.GetFeatureCount()
+
+    # deleting shapefiles after test
+    driver.DeleteDataSource(input_path)
+    driver.DeleteDataSource(method_path)
+    driver.DeleteDataSource(output_path)
+
+    assert featureCount == 2
+
+
+###############################################################################
+
+# Test Union
+
+
+def test_ogr_layer_algebra_2():
```

Comment:
```suggestion
def test_ogr_layer_algebra_union():
```

Reply:
Changed in https://github.com/kssvrk/gdal/commit/841f3fe4b409219623527f0a021c34eae6071697